DDOC-496: New user avatar endpoints (#177)
box/box-developer-changelog: content/2022/05-18-new-endpoints-for-user-avatar.md

---
applied_at: "2022-05-18"
applies_to: 
- api
is_impactful: true
is_new_feature: true
collapse: true
show_excerpt: true
release_source_url: ''
---

# New Endpoints to Manage User Avatars

> The User Avatar API now supports adding, updating, and deleting user avatars.

Reviewer: ```suggestion
Starting today, the User Avatar API supports adding, updating, and deleting user avatars.
```

> To create or update a user avatar, call the [`POST /user-avatar`][2] endpoint. Remember to include the path to the file you want to upload.

Reviewer: ```suggestion
To create or update a user avatar, call the [`POST /user-avatar`][2] endpoint. Remember to include the path to the file you want to upload. If an avatar already exists, it will be replaced with the new photo uploaded.
```

```sh

> curl -i -X POST -L https://api.box.net/2.0/users/userID/avatar --H 'Authorization: Bearer {token}'--form 'pic=@"path/to/file/file.jpeg"'

Reviewer: ```suggestion
curl -i -X POST -L https://api.box.net/2.0/users/userID/avatar --H 'Authorization: Bearer <ACCESS TOKEN>'--form 'pic=@"path/to/file/file.jpeg"'
```

```
The response includes the [user avatar][4] object with URLs to the file location.

To delete a user avatar, call the [`DELETE /user-avatar`][3] endpoint:

```sh

> curl -i -X DELETE -L https://api.box.net/2.0/users/userID/avatar -H 'Authorization: Bearer {token}'

Reviewer: ```suggestion
curl -i -X DELETE -L https://api.box.net/2.0/users/userID/avatar -H 'Authorization: Bearer <ACCESS TOKEN>'
```
